azurerm/internal/services/datafactory/data_factory.go

```diff
@@ -283,9 +283,25 @@ func expandDataFactoryDatasetLocation(d *pluginsdk.ResourceData) datafactory.Bas
 		return expandDataFactoryDatasetAzureBlobFSLocation(d)
 	}
 
+	if _, ok := d.GetOk("sftp_server_location"); ok {
+		return expandDataFactoryDatasetSFTPServerLocation(d)
+	}
+
 	return nil
 }
 
+func expandDataFactoryDatasetSFTPServerLocation(d *pluginsdk.ResourceData) datafactory.BasicDatasetLocation {
+	props := d.Get("sftp_server_location").([]interface{})[0].(map[string]interface{})
```

Comment:
I think it's better to have an empty check `d.Get("sftp_server_location").([]interface{})`

Reply:
Done